src/pages/default/connections/AddConnectionScreen.tsx

```diff
@@ -0,0 +1,323 @@
+    } catch (e: any) {
+      alert(e.message);
+      navigation.goBack();
+    }
+  };
+
+  const extractQueryParams = (qrContent: string) => {
+    const { query } = queryString.parseUrl(qrContent);
+
+    if (
+      typeof query.appDid !== "string" ||
+      typeof query.url !== "string" ||
+      typeof query.nonce !== "string"
+    ) {
+      throw new Error(
+        "Received a malformed QR code. Please try scanning again."
+      );
+    }
+
+    const dwaDID = query.appDid;
+    const serverURL = query.url;
+    const connectNonce = stringifiedArrayToUInt8Array(query.nonce);
+
+    if (!(connectNonce instanceof Uint8Array)) {
+      throw new Error(
+        "Not a valid connect nonce. Failed to create Uint8Array from scanned string."
+      );
+    }
+
+    return { dwaDID, connectNonce, serverURL };
+  };
+
+  const initConnect = (
+    dwaDID: string,
+    connectNonce: Uint8Array,
+    serverURL: string
+  ) => {
+    const dwaSignKeyID = dwaDID.substring(dwaDID.lastIndexOf(":") + 1);
+    const dwaSignPublicKey = base58btcMultibaseToBytes(dwaSignKeyID);
+
+    // derive the Connect Id from the DWA's signing public key.
+    const connectIdU8A = hkdf(
+      sha256,
+      dwaSignPublicKey,
+      undefined,
+      "connect_uuid",
+      32
+    );
+    const connectId = base64url.baseEncode(connectIdU8A);
+
+    // derive the Connect Key from the DWA's signing public key
+    const connectKey = hkdf(
+      sha256,
+      dwaSignPublicKey,
+      undefined,
+      "connect_encryption_key",
+      32
+    );
+
+    console.log(dwaSignKeyID);
+    console.log(dwaSignPublicKey);
+    console.log(connectIdU8A);
+    console.log(connectId);
+
+    // call into DWN server using the passed QR data
+    const connectionRequestCipherText = fetchEncryptedConnectionRequest(
+      connectId,
+      serverURL
+    );
+
+    // decrypt the cipher text using the connectKey and connectNonce
+    const decryptedPermissionsRequest = decryptConnectionRequest(
+      connectionRequestCipherText,
+      connectKey,
+      connectNonce
+    );
+
+    // TODO: raise permissions request UI? Accept/deny. need designs.
+    Alert.alert(
+      "Permissions Request",
+      `DWA would like access to ${JSON.stringify(decryptedPermissionsRequest)}`,
+      [
+        {
+          text: "Cancel",
+          onPress: () => navigation.goBack(),
+          style: "cancel",
+        },
+        { text: "OK", onPress: () => navigation.goBack() },
+      ]
+    );
+
+    NOOP(decryptedPermissionsRequest);
+
+    // TODO: raise identity selection UI. need designs.
+    const selectedIdentityDid = "did:ion:EiCabc123";
+
+    // after User clicks "accept." Send grant? (check terminology) to DWN server.
+    const data = postPermissionsAuthorization(
+      dwaSignPublicKey,
+      selectedIdentityDid
+    );
+
+    // TODO: Add connection data to the DWN level store?
+    addConnectionToDwn(data);
```

Comment:
Should this "success" logic get pulled out of init? Totally fine if this is just placeholder for now though.

Reply:
Yeah I'll address this as soon as I modify the design